File: src/main/java/com/example/common/entity/ElementCoordinates.java
package com.example.common.entity;

import com.example.common.Rotation;

import javax.persistence.*;

/**
 * Created by Anatol on 11.06.16.
 */
@Entity
@Table(name = "elementCoordinates")
public class ElementCoordinates extends  BaseEntity {
    @Column(nullable = false)
    private Long schemeId;
    @OneToOne
    @JoinColumn(name = "element_id", nullable = false)
    private Element element;
    @Column(nullable = false)
    private Long xCoordinate;
    @Column(nullable = false)
    private Long yCoordinate;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Rotation rotation = Rotation.ROTATION_RIGHT;

    public Rotation getRotation() {
        return rotation;
    }

    public void setRotation(Rotation rotation) {
        this.rotation = rotation;
    }

    public Long getSchemeId() {
        return schemeId;
    }

    public void setSchemeId(Long schemeId) {
        this.schemeId = schemeId;
    }

    public Element getElement() {
        return element;
    }

    public void setElement(Element element) {
        this.element = element;
    }

    public Long getxCoordinate() {
        return xCoordinate;
    }

    public Long setxCoordinate(Long xCoordinate) {
        this.xCoordinate = xCoordinate;
        return xCoordinate;
    }

    public Long getyCoordinate() {
        return yCoordinate;
    }

    public Long setyCoordinate(Long yCoordinate) {
        this.yCoordinate = yCoordinate;
        return yCoordinate;
    }
}

File: frontend/src/app/dto/ElementCoordinates.ts
import {Element} from "./Element";
export default class ElementCoordinates {
    private id: number;
    private schemeId: number;
    private element: Element;
    private xCoordinate: number;
    private yCoordinate: number;
    private rotation: string;
    private text: string;

    getText():string{
      return this.text;
      }

  setText(value:string){
      this.text=value;
      }

  getId():number{
      return this.id;
      }

  setId(value:number){
      this.id=value;
      }
  getSchemeId():number{
    return this.schemeId;
  }

  setSchemeId(value:number){
    this.schemeId=value;
  }

  getElement():Element{
      return this.element;
      }

  setElement(value:Element){
      this.element=value;
      }

  getXCoordinate():number{
      return this.xCoordinate;
      }

  setXCoordinate(value:number){
      this.xCoordinate=value;
      }

  getYCoordinate():number{
      return this.yCoordinate;
      }

  setYCoordinate(value:number){
      this.yCoordinate=value;
      }

  getRotation():string{
      return this.rotation;
      }

  setRotation(value:string){
      this.rotation=value;
      }
}
